### DISCUSSION TITLE
What is correct way to read the input and store into a variable?

### DISCUSSION THREAD

**Discussion by ipa-rar** [zauberzeug/nicegui#1453] [Q&A] [answered]
### Question

I am trying to read some input from the user and assign it into some variables to send a API request. However I think there is something wrong in my implementation because of which I am unable to use the user input. What would be the correct way to get the user input and then use the input values ?

_This is my nicegui input form_ 
![image](https://github.com/zauberzeug/nicegui/assets/49311464/e803141e-864a-4398-aa9e-216723b44a63)

_Code_
```python
@ui.page('/ui')
  def view():
        def update_payload(name, value):
            payload[name] = value

        with ui.card().classes('my-5 h-40'):
            ui.label('Select Test').classes('text-h6')
            with ui.element('div').classes('p-3 bg-blue-100'):
                select0 = ui.select(TESTCASES, value=TESTCASES[0])
                update_payload("test_name", select0.value)
        with ui.card().classes('my-0 h-40'):
            ui.label('Select the tracking object').classes(
                'text-h6')
            ui.radio(['tracker_1', 'tracker_2'],value='tracker_1', on_change=lambda e: update_payload("tracking_object", str(e.value))).props('inline')
      
        with ui.card().classes('my-0 h-120'):
            ui.label('Set environment conditions').classes(
                'text-h6')
            for field_name in ID_INPUT:
                ui.input(field_name, on_change=lambda e: update_payload(field_name, str(e.value)))
      
            for field_name in FLOAT_INPUT:
                ui.input(field_name, on_change=lambda e: update_payload(field_name, float(e.value)))
```

**Comment by Anindya088** [comment, marked as answer]
Hi @ipa-rar ,

I suppose you want to retrieve the values that users write into the ```ui.input``` boxes.
In that case, you can simply use ```bind_text``` properties to assign the ```'value'``` to other elements.

Something like this:
```python
username = ui.input('username')
label = ui.label().bind_text_from(username, 'value')
```

Or otherwise, you can also assign it to another element and use ```.value``` property using a lambda function. 
Or assign to an internal variable for other usage (below example I show print)
For ex: something like this
```python
username = ui.input('username')
ui.button('Show Username',on_click=lambda: show_username())
label = ui.label()

def show_username():
    label.text = username.value
    internal_var = username.value
    print(internal_var)

```

Thanks,
Anindya
